### DISCUSSION TITLE
SchemaChange Key-Pair Authentication

### DISCUSSION THREAD

**Discussion by hy-kipi** [Snowflake-Labs/schemachange#306] [Q&A]
I am using GitHub Actions & Schemachange for Snowflake. How can I use Key-Pair for Authentication?

**Comment by Igladyshev** [comment]
I guess the original post is not relevant, but I dealt with the same issue and finally figured it out - it is all because of discrepancy in connection config variables naming, especially in the private key authentication part. The following is how I made it work:

1. Create your `connections.toml` file as the following:
`
[default]
account = "$SNOWFLAKE_ACCOUNT"
user = "$SNOWFLAKE_USER"
database = "$SNOWFLAKE_DB"
schema = "$SNOWFLAKE_SCHEMA"
warehouse = "$SNOWFLAKE_WAREHOUSE"
role = "$SNOWFLAKE_ROLE"
authenticator = "SNOWFLAKE_JWT"
private_key_file_pwd = $QUOTED_PASSPHRASE
private_key_file = "$(pwd)/rsa_key.p8"
`
2. When you call the `schemachange` use the following command:

`
#!/bin/zsh
export CONFIG_FOLDER="$(pwd)/migrations"
export ROOT_FOLDER="$CONFIG_FOLDER/changes"
CONNECTIONS_FILE_PATH="$CONFIG_FOLDER/connections.toml"
echo $CONNECTIONS_FILE_PATH
schemachange --config-folder "$CONFIG_FOLDER" -f "$ROOT_FOLDER" --connections-file-path "$CONNECTIONS_FILE_PATH" --connection-name default -c YOUR_DATABASE.SCHEMACHANGE.CHANGE_HISTORY
`

This way it works for me

**Comment by sfc-gh-tmathew** [comment]
@Igladyshev please try release 4.1.0 and see if your situation gets simplified and predictable with 4.1.0.

@hy-kipi Please refere to updated examples and let us know if that helps.

**Reply by Igladyshev** [reply]
I upgraded to 4.1.0 and I like the change!
